app/scripts/modules/core/src/pipeline/config/triggers/trigger.directive.js

```diff
@@ -40,6 +41,15 @@ module.exports = angular.module('spinnaker.core.pipeline.config.trigger.triggerD
       return artifact.type && artifact.name;
     };
 
+    this.summarizeExpectedArtifacts = function() {
```

Comment:
Does `expected.summary` end up in the pipeline config? Instead of mutating the artifacts, you could just a define a function that formats the artifact:

```
<ui-select-choices ...> 
   <span>{{ triggerCtrl.myExpectedArtifactFormatter(expected) }}</span>
</ui-select-choices/>
```